lib/cdo.rb

```diff
@@ -110,11 +110,13 @@ def hourofcode_hostname
     end
 
     def codeprojects_hostname
-      canonical_hostname('codeprojects.org')
+      return 'codeprojects.org' if rack_env?(:production)
+      return "localhost.codeprojects.org" if rack_env?(:development) || ci_webserver?
+      return "#{stack_name}.codeprojects.org"
     end
```

Comment:
This is definitely a change of behavior for adhocs, which were previously returning `adhoc.codeprojects.org`. They will now return something like `adhoc-darin-preview-codeprojects.codeprojects.org`. In staging and test, there should be no change, and in prod, we explicitly use the apex `codeprojects.org`